pkg/pushsync/pushsync.go

```diff
@@ -448,20 +430,14 @@ func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk, origin bo
 
 			ps.measurePushPeer(result.pushTime, result.err, origin)
 
-			if ps.warmedUp() && !errors.Is(result.err, accounting.ErrOverdraft) {
-				ps.skipList.Add(ch.Address(), result.peer, sanctionWait)
-			}
-
 			if result.err == nil {
 				return result.receipt, nil
 			}
 
 			ps.metrics.TotalFailedSendAttempts.Inc()
 			logger.Debug("could not push to peer", "peer_address", result.peer, "error", result.err)
 
-			if result.sent {
-				sentErrorsLeft--
-			}
+			sentErrorsLeft--
 
 			retry()
```

Comment:
what is reason for doing `retry()` here? it seems that after result is handled it shouldn't retry to push to closes peer?